qa/tasks/deepsea.py

```diff
@@ -1697,6 +1701,35 @@ def _repositories_to_remote(self, remote):
             args=args
             )
 
+    def _patch(self, **kwargs):
+        """Perform zypper patch with maintenance test repos from 'deepsea:patch' section
+
+        It will target all nodes in the cluster.
+
+        Repository format provided via deepsea-overrides.yaml in teuthology-suite.
+
+        deepsea:
+            patch:
+              - url: http://<server>/artifacts/ci/XYZ/SUSE_Updates_Storage_5_x86_64
+                name: SUSE_Maintenance_13609
+                prio: 1
+        """
+        if self.patch_repositories:
+            for repo in self.patch_repositories:
+                cmd = (
+                    "zypper -n addrepo --refresh --no-gpgcheck {url} {name} && "
+                    "zypper -n --no-gpg-checks refresh && "
+                    "zypper list-patches --with-optional --repo {name} && "
+                    "zypper -n patch -l -y --details --with-optional --repo {name} && "
+                    "zypper ps -s"
+                    .format(url=repo['url'], name=repo['name'])
+                )
+                self.ctx.cluster.run(args=[
+                    'sudo', 'sh', '-c', cmd
+                ])
+        else:
+            self.log.warning("No patch repositories found.")
```

Comment:
Should we error out here? I think it's an error if we want to test a maint update and no patch repos are found. Isn't it?

Reply:
Already discussed on demo meeting.